java/dataset/src/main/java/org/apache/arrow/dataset/scanner/FragmentScanOptions.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.arrow.dataset.scanner;
+
+import java.util.Map;
+
+public interface FragmentScanOptions {
+  String typeName();
+
+  int fileFormatId();
+
+  String[] serialize();
+
+  /**
+   * Serialize the map to string array.
+   *
+   * @param config config map
+   * @return string array for serialization
+   */
+  default String[] serializeMap(Map<String, String> config) {
```

Comment:
Are we actually doing any serialization here or just converting a map to a String[] such that even indexes are keys and odd indexes are values? Maybe a different name?

Reply:
just converting

Reply:
Then, maybe `convertMapToStringArray` or something?